dbms/src/Functions/FunctionsStringSimilarity.cpp

```diff
@@ -281,18 +276,91 @@ struct NgramDistanceImpl
         }
     }
 
+    static void vector_vector(
+        const ColumnString::Chars & haystack_data,
+        const ColumnString::Offsets & haystack_offsets,
+        const ColumnString::Chars & needle_data,
+        const ColumnString::Offsets & needle_offsets,
+        PaddedPODArray<Float32> & res)
+    {
+        const size_t haystack_offsets_size = haystack_offsets.size();
+        size_t prev_haystack_offset = 0;
+        size_t prev_needle_offset = 0;
+
+        NgramStats common_stats;
+        memset(common_stats, 0, sizeof(common_stats));
```

Comment:
You can just write 
```
NgramStats common_stats{};
```

It is named "zero initialization" in C++.

Reply:
Fixed in master